libsmtutil/SMTLib2Context.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsmtutil/SMTLib2Context.h>
+
+#include <boost/functional/hash.hpp>
+
+#include <range/v3/algorithm/find_if.hpp>
+
+namespace solidity::smtutil
+{
+
+std::size_t SortPairHash::operator()(std::pair<SortId, SortId> const& pair) const
+{
+	std::size_t seed = 0;
+	boost::hash_combine(seed, pair.first.id);
+	boost::hash_combine(seed, pair.second.id);
+	return seed;
+}
+
+SMTLib2Context::SMTLib2Context()
+{
+	clear();
+}
+
+bool SMTLib2Context::isDeclared(std::string const& _name) const
+{
+	return m_functions.count(_name) > 0;
+}
+
+void SMTLib2Context::setDeclared(std::string const& _name, SortPointer const& _sort)
```

Comment:
`declare` name would be better. `setDeclared` looks like we set a sort of an already declared function.

Reply:
Fixed.